homeassistant/components/moehlenhoff_alpha2/__init__.py

```diff
@@ -0,0 +1,124 @@
+"""Support for the Moehlenhoff Alpha2."""
+import asyncio
+import logging
+import time
+
+from moehlenhoff_alpha2 import Alpha2Base
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+class CannotConnect(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up Alpha2 integration."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up a config entry."""
+    base = Alpha2Base(entry.data["host"])
+    try:
+        await base.update_data()
+    except TimeoutError as err:
+        raise exceptions.ConfigEntryNotReady from err
+    base_uh = Alpha2BaseUpdateHandler(base, 60)
+    hass.data.setdefault(DOMAIN, {"connections": {}, "devices": set()})
+    hass.data[DOMAIN]["connections"][entry.entry_id] = base_uh
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok and entry.entry_id in hass.data[DOMAIN]:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry):
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+class Alpha2BaseUpdateHandler:
+    """Keep the base instance in one place and centralize the update."""
+
+    def __init__(self, base, scan_interval):
+        """Initialize the base handle."""
+        self.base = base
+        self.scan_interval = scan_interval
+        self._last_update = 0
+        self._heatarea_update_callbacks = {}
+        self._loop = asyncio.get_event_loop()
+
+    def add_heatarea_update_callback(self, callback, heatarea_nr):
+        """Add a callback which will be run when data of the given heatarea is updated."""
+        if heatarea_nr not in self._heatarea_update_callbacks:
+            self._heatarea_update_callbacks[heatarea_nr] = []
+        self._heatarea_update_callbacks[heatarea_nr].append(callback)
+
+    async def async_update(self):
+        """Pull the latest data from the Alpha2 base."""
+        # Only update every update_interval
+        if (time.monotonic() - self._last_update) >= self.scan_interval:
```

Comment:
it's probably easier for a case like this to schedule the next update after this update completed, using ```from homeassistant.helpers.event import async_call_later```, then you also don't have to worry about storing last_update

Reply:
Using homeassistant.helpers.event.async_track_time_interval now.